server/graphql/v1/CollectiveInterface.js

```diff
@@ -650,7 +650,7 @@ export const CollectiveInterfaceType = new GraphQLInterfaceType({
       },
       data: {
         type: GraphQLJSON,
-        deprecationReason: '2020-10-08: This field is not provided anymore and will return an empty object',
+        description: 'Data related to events like privateInstructions',
```

Comment:
:warning: We should keep this deprecation. We don't want to expose the full collective `data`, especially as it might contain some private information that is not meant to be exposed. We should rather expose a new GraphQL field like `privateInstructions`, that exposes just this information.

The permission check also needs to be a bit more permissive: all event participants must be allowed to see this. We need something like this (needs to be tested):
```es6
if (collective.type === types.EVENT && (
	req.remoteUser?.isAdminOfCollective(collective)
	|| req.remoteUser?.hasRole(roles.PARTICIPANT, collective)
) {
  return collective.data?.privateInstructions
}
```